src/lib/remark-plugins/remark-image-dimensions/index.ts

```diff
@@ -0,0 +1,71 @@
+import type { Image, Root } from 'mdast'
+import { Plugin, Transformer } from 'unified'
+import { visit } from 'unist-util-visit'
+import probe from 'probe-image-size'
+
+/**
+ * This plugin injects image dimensions for images whose src references the mktg-content-api.
+ * This only applies to tutorials currently, for docs content this is done via mktg-content-workflows
+ * https://github.com/hashicorp/mktg-content-workflows/blob/main/shared/transforms/rewrite-site-static-assets.ts
+ */
+export const remarkPluginInjectImageDimensions: Plugin = (): Transformer => {
+	return async function transformer(tree: Root) {
+		const imageNodesForDimensions = []
+		visit(tree, 'image', (node: Image) => {
+			/**
+			 * Only transform nodes using the mktg-content-api src.
+			 * This is dependent on `rewriteStaticAssets` running before this plugin
+			 * to transform the src URLs from local file paths to mktg-content-api urls
+			 * See: src/lib/remark-plugins/rewrite-static-tutorials-assets/index.ts
+			 */
+			if (node.url.startsWith(process.env.MKTG_CONTENT_API)) {
+				imageNodesForDimensions.push(node)
+			}
+		})
+		/**
+		 * If width and height aren't defined via props by the author, we attempt
+		 * to calculate the file dimensions and append those props to the source string
+		 *
+		 * This involves async tasks so is handled in a separate loop from the 'visit'
+		 * where async isn't supported. Taken from suggestion in this issue
+		 *  https://github.com/syntax-tree/unist-util-visit-parents/issues/8#issuecomment-1413405543
+		 */
+		for (const node of imageNodesForDimensions) {
+			const url = await getUrlWithDimensions(node.url)
+
+			if (url) {
+				node.url = url
+			}
+		}
```

Comment:
(Build optimization nit): Consider using `await Promise.all()` here so we can avoid waiting on sequential image "probe" calls. 

Something like... (pseudocode, not guaranteed to work)

```ts
const promises = imageNodesForDimensions.map(async (node) => {
  const url = await getUrlWithDimensions(node.url)
  if (url) {
    node.url = url
  }
})
await Promise.all(promises)
```

Rationale: This is because remarkPlugins get called during build time for any statically generated pages (ie. docs n tutorials) so if the remarkPlugins slow down, there will be perf implications at **build** and **regeneration** time

Reply:
Yes, great suggestion 💯